### PR TITLE
Added !cve to VulnDB

### PR DESCRIPTION
**Pull request by hod-alpert**
## Status
- [] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://github.com/demisto/etc/issues/20984

## Does it break backward compatibility?
   - [ ] No

## Must have
- [x] Tests
- [ ] Documentation 

## Technical Writer Review
- [ ] [CHANGELOG](https://github.com/demisto/content/edit/hod/cve_indicator/Integrations/VulnDB/CHANGELOG.md?pr=%2Fdemisto%2Fcontent%2Fpull%2F5517)
- [ ] [YAML](https://github.com/demisto/content/edit/hod/cve_indicator/Integrations/VulnDB/VulnDB.yml?pr=%2Fdemisto%2Fcontent%2Fpull%2F5517)

### CODE REVIEW

**Comment by hod-alpert** [comment]
@kirbles19 Please review yml file for !cve command and changlog file

File: Integrations/VulnDB/VulnDB.py [hunk 1]
```diff
@@ -117,41 +122,32 @@ def vulndb_vulnerability_to_entry(vuln):
...
-            }
-
-            human_readable = tableToMarkdown(f'Result for vulnerability ID: {ec["VulnDB"]["Vulnerability"]["ID"]}', {
-                'Title': ec['VulnDB']['Vulnerability']['Title'],
-                'Description': ec['VulnDB']['Vulnerability']['Description'],
-                'Publish Date': ec['VulnDB']['Vulnerability']['PublishedDate'],
-                'Solution Date': ec['VulnDB']['Vulnerability']['SolutionDate']
-            })
+        return_error('No "vulnerability" or "results" keys in the returned JSON')
+        return
```
Comment (avidan-H) [inline Integrations/VulnDB/VulnDB.py]: You can remove this `return`, `return_error` will `sys.exit`.

File: Integrations/VulnDB/VulnDB.py [hunk 2]
```diff
@@ -117,41 +122,32 @@ def vulndb_vulnerability_to_entry(vuln):
...
-                'EntryContext': ec
-            })
+        demisto.results({
+            'Type': entryTypes['note'],
+            'Contents': res,
+            'ContentsFormat': formats['json'],
+            'HumanReadable': human_readable,
+            'HumanReadableFormat': formats['markdown'],
+            'EntryContext': ec
+        })
```
Comment (avidan-H) [inline Integrations/VulnDB/VulnDB.py]:
```suggestion
        return_outputs(readable_output=human_readable, outputs=ec, raw_response=res)
```
You can replace the `demisto.results()` code block with the above.

File: Integrations/VulnDB/VulnDB.py [hunk 3]
```diff
@@ -432,3 +452,5 @@ def vulndb_get_version_command():
     vulndb_get_version_command()
 elif demisto.command() == 'vulndb-get-updates-by-dates-or-hours':
     vulndb_get_updates_by_dates_or_hours_command()
+elif demisto.command() == 'cve':
+    vulndb_get_cve()
```
Comment (avidan-H) [inline Integrations/VulnDB/VulnDB.py]:
```suggestion
    vulndb_get_cve_command()
```

File: Integrations/VulnDB/VulnDB.yml [hunk 4]
```diff
@@ -141,13 +141,15 @@ script:
       secret: false
     - default: false
       defaultValue: '20'
-      description: Maximum number of entries to return. A high number of entries might effect performance.
+      description: Maximum number of entries to return. A high number of entries might
+        effect performance.
```
Comment (avidan-H) [inline Integrations/VulnDB/VulnDB.yml]:
```suggestion
        affect performance.
```

File: Integrations/VulnDB/VulnDB.yml [hunk 5]
```diff
@@ -178,7 +180,8 @@ script:
...
```
Comment (avidan-H) [inline Integrations/VulnDB/VulnDB.yml]:
```suggestion
        affect performance.
```

File: Integrations/VulnDB/VulnDB.yml [hunk 6]
```diff
@@ -209,7 +212,8 @@ script:
...
```
Comment (avidan-H) [inline Integrations/VulnDB/VulnDB.yml]:
```suggestion
        affect performance.
```

File: Integrations/VulnDB/VulnDB.yml [hunk 7]
```diff
@@ -246,7 +250,8 @@ script:
...
```
Comment (avidan-H) [inline Integrations/VulnDB/VulnDB.yml]:
```suggestion
        affect performance.
```

File: Integrations/VulnDB/VulnDB.yml [hunk 8]
```diff
@@ -355,14 +360,15 @@ script:
...
```
Comment (avidan-H) [inline Integrations/VulnDB/VulnDB.yml]:
```suggestion
        affect performance.
```

File: Integrations/VulnDB/VulnDB.yml [hunk 9]
```diff
@@ -141,13 +141,15 @@ script:
...
       isArray: false
       name: max_size
       required: false
       secret: false
     deprecated: false
-    description: Returns all, or specific, vendor details to include vendor name and ID.
+    description: Returns all, or specific, vendor details to include vendor name and
+      ID.
```
Comment (avidan-H) [inline Integrations/VulnDB/VulnDB.yml]: This description isn't so clear. If you could try and make it better that would be great, but either way, a technical writer will review so if you want to wait for their review, that's also ok.

File: Integrations/VulnDB/VulnDB.yml [hunk 10]
```diff
@@ -465,14 +471,15 @@ script:
       secret: false
     - default: false
       defaultValue: '20'
-      description: Maximum number of entries to return. A high number of entries might effect performance.
+      description: Maximum number of entries to return. A high number of entries might
+        effect performance.
```
Comment (avidan-H) [inline Integrations/VulnDB/VulnDB.yml]:
```suggestion
        affect performance.
```
Make this change wherever it's relevant.

File: Integrations/VulnDB/VulnDB.py [hunk 11]
```diff
@@ -404,6 +400,30 @@ def vulndb_get_version_command():
     vulndb_product_results_to_demisto_results(res)
 
 
+def vulndb_get_cve():
+    cve_id = demisto.args()['cve_id']
```
[the remark this answers is not in the document]
Comment (hod-alpert) [reply Integrations/VulnDB/VulnDB.py]: It's mandatory, so I expect it to be there.

File: Integrations/VulnDB/VulnDB.yml (hunk 9 above)
Comment (hod-alpert) [reply Integrations/VulnDB/VulnDB.yml]: This is actually not my doing and I'm not sure what this API does, it must have been altered by accident.

As for now I don't really know what this API should do, do you want me to dig deeper into it?

In the meantime- I returned it to the original

**Comment by avidan-H** [comment]
Need to update the `display` field for the `insecure` parameter.
image

**Comment by avidan-H** [comment]
You also need to fix your test playbook it seems like. Then you can merge.

File: Integrations/VulnDB/VulnDB.yml [hunk 12]
```diff
@@ -246,7 +246,7 @@ script:
       secret: false
     - default: false
       defaultValue: '20'
-      description: Maximum number of entries to return. A high number of entries might effect performance.
+      description: Maximum number of entries to return. A high number of entries might affect performance.
```
Comment (kirbles19) [inline Integrations/VulnDB/VulnDB.yml]: Maximum number of entries to return. A high number of entries might affect performance. The default value is "20".

File: Integrations/VulnDB/VulnDB.yml [hunk 13]
```diff
@@ -209,7 +209,7 @@ script:
...
```
Comment (kirbles19) [inline Integrations/VulnDB/VulnDB.yml]: Maximum number of entries to return. A high number of entries might affect performance. The default value is "20".

File: Integrations/VulnDB/VulnDB.yml [hunk 14]
```diff
@@ -178,7 +178,7 @@ script:
...
```
Comment (kirbles19) [inline Integrations/VulnDB/VulnDB.yml]: Maximum number of entries to return. A high number of entries might affect performance. The default value is "20".

File: Integrations/VulnDB/VulnDB.yml [hunk 15]
```diff
@@ -141,7 +141,7 @@ script:
...
```
Comment (kirbles19) [inline Integrations/VulnDB/VulnDB.yml]: Maximum number of entries to return. A high number of entries might affect performance. The default value is "20".

File: Integrations/VulnDB/VulnDB.yml [hunk 16]
```diff
@@ -355,7 +355,7 @@ script:
...
```
Comment (kirbles19) [inline Integrations/VulnDB/VulnDB.yml]: Maximum number of entries to return. A high number of entries might affect performance. The default value is "20".

File: Integrations/VulnDB/VulnDB.yml [hunk 17]
```diff
@@ -465,7 +465,7 @@ script:
...
```
Comment (kirbles19) [inline Integrations/VulnDB/VulnDB.yml]: Maximum number of entries to return. A high number of entries might affect performance. The default value is "20".

File: Integrations/VulnDB/VulnDB.yml [hunk 18]
```diff
@@ -569,7 +569,7 @@ script:
...
```
Comment (kirbles19) [inline Integrations/VulnDB/VulnDB.yml]: Maximum number of entries to return. A high number of entries might affect performance. The default value is "20".

File: Integrations/VulnDB/VulnDB.yml [hunk 19]
```diff
@@ -672,7 +672,7 @@ script:
...
```
Comment (kirbles19) [inline Integrations/VulnDB/VulnDB.yml]: Maximum number of entries to return. A high number of entries might affect performance. The default value is "20".

File: Integrations/VulnDB/VulnDB.yml [hunk 20]
```diff
@@ -775,7 +775,7 @@ script:
...
```
Comment (kirbles19) [inline Integrations/VulnDB/VulnDB.yml]: Maximum number of entries to return. A high number of entries might affect performance. The default value is "20".

**Comment by kirbles19** [comment]
@hod-alpert Done. The command looks great (for docs)!

**Pull request merged**